src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/transpose_matmul_fusion.cpp

```diff
@@ -0,0 +1,93 @@
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<TransposeOrderParams> obj) {
+        std::vector<int64_t> target_order;
+        ov::element::Type input_precision;
+
+        std::tie(target_order, input_precision) = obj.param;
+
+        std::ostringstream result;
+        result << "transpose_order=[";
+        for (const auto& order : target_order) {
+            result << order << "_";
+        }
+        result << "]_input_precision=" << input_precision;
+        return result.str();
+    }
+
+protected:
+    std::shared_ptr<ov::Model> init_subgraph(ov::element::Type& input_precision, const std::vector<int64_t>& target_transpose_order) {
+        ov::PartialShape input_a_shape = ov::PartialShape{-1, -1, -1, -1};
+        ov::PartialShape input_b_shape = ov::PartialShape{-1, -1};
+
+        auto input_a = std::make_shared<ov::op::v0::Parameter>(input_precision, input_a_shape);
+        auto input_b = std::make_shared<ov::op::v0::Parameter>(input_precision, input_b_shape);
+
+        auto transpose_order_a = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{target_transpose_order.size()}, target_transpose_order);
+        auto transpose_a = std::make_shared<ov::op::v1::Transpose>(input_a, transpose_order_a);
+
+        auto matmul = std::make_shared<ov::op::v0::MatMul>(transpose_a, input_b, false, false);
+
+        auto model = std::make_shared<ov::Model>(ov::NodeVector{matmul}, ov::ParameterVector{input_a, input_b});
+        return model;
+    }
+
+private:
+    ov::element::Type input_precision = ov::element::f16;
+    std::vector<int64_t> order = {0, 1, 2, 3};
+};
+
+TEST_P(TransposeMatmulFuseTest, smoke_allowed_transpose_order) {
+    std::vector<int64_t> target_order;
+    ov::element::Type input_precision;
+    std::tie(target_order, input_precision) = GetParam();
+    auto function = init_subgraph(input_precision, target_order);
+
+    std::string targetDevice = ov::test::utils::DEVICE_GPU;
+    ov::Shape input_a_shape = {5, 6, 7, 8};
+    ov::Shape input_b_shape = {8, 7};
+    ov::Shape permuted_shape_b(input_b_shape.size());
+    for (size_t i = 0; i < permuted_shape_b.size(); ++i) {
+        permuted_shape_b[i] = input_a_shape[target_order[target_order.size() - 1 - i]];
+    }
+
+    auto input_tensor_a = ov::test::utils::create_and_fill_tensor(input_precision, input_a_shape, 0.0f, 1.0f);
+    auto input_tensor_b = ov::test::utils::create_and_fill_tensor(input_precision, permuted_shape_b, 0.0f, 1.0f);
+
+    auto core = ov::test::utils::PluginCache::get().core();
+    ov::CompiledModel cM = core->compile_model(function, targetDevice, {ov::hint::inference_precision(input_precision)});
+    auto request = cM.create_infer_request();
+    request.set_input_tensor(0, ov::Tensor(input_precision, input_a_shape, input_tensor_a.data()));
+    request.set_input_tensor(1, ov::Tensor(input_precision, permuted_shape_b, input_tensor_b.data()));
+    request.infer();
+}
+
```

Comment:
maybe also need to add test to here also (supports_immad to true/false): src/plugins/intel_gpu/tests/unit/transformations/transpose_matmul_fusion_test.cpp